api/src/services/applications/create_application_attachment.py

```diff
@@ -51,22 +78,22 @@ def create_application_attachment(
     # If we don't do this, we see `Can't attach instance of 'User' to session` errors.
     user = db_session.merge(user)
 
-    # Create the application attachment
-    application_attachment = ApplicationAttachment(
-        application_attachment_id=application_attachment_id,
-        application=application,
-        file_location=s3_file_location,
-        # In the file_name column we store the users actual file name unmodified
-        # so when we display it on the UI it matches whatever they uploaded.
-        file_name=file_util.get_file_name(file_attachment.filename),
-        mime_type=file_attachment.mimetype,
-        file_size_bytes=file_size_bytes,
-        user=user,
-    )
-    db_session.add(application_attachment)
+    # Set or update the application attachment properties
+    application_attachment.application = application
+    application_attachment.file_location = s3_file_location
+    # In the file_name column we store the users actual file name unmodified
+    # so when we display it on the UI it matches whatever they uploaded.
+    application_attachment.file_name = file_util.get_file_name(file_attachment.filename)
+    application_attachment.mime_type = file_attachment.mimetype
+    application_attachment.file_size_bytes = file_size_bytes
+    application_attachment.user = user
+
+    # Only add to session if it's a new attachment (checking if it already exists in the session)
+    if application_attachment not in db_session:
```

Comment:
I think you're safe to just always do `db_session.add(..)` on something already in the session since it really just changes internal state to the same one as it would be when we fetch it from the DB.